lib/collection/src/update_handler.rs

```diff
@@ -352,4 +341,50 @@ impl UpdateHandler {
             .send(OptimizerSignal::Stop)
             .unwrap_or_else(|_| debug!("Optimizer already stopped"));
     }
+
+    async fn flush_worker(
+        segments: LockedSegmentHolder,
+        wal: Arc<Mutex<SerdeWal<CollectionUpdateOperations>>>,
+        flush_interval_sec: u64,
+        mut stop_receiver: oneshot::Receiver<()>,
+    ) {
+        loop {
+            // Stop flush worker on signal or if sender was dropped
+            // Even if timer did not finish
+            tokio::select! {
+                _ = tokio::time::sleep(Duration::from_secs(flush_interval_sec)) => {},
+                _ = &mut stop_receiver => {
```

Comment:
is there an implicit `stop_receiver.await` here?

Reply:
The process is a bit more difficult than `await` - it polls the futures listed here by itself

Reply:
But looking from the high level - yes - one of them will be "awaited" to completion.

Reply:
Got it, just checked the docs, it seems legit :+1: